specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+  const backEndResult = useBackendApi();
+  return React.useMemo(
+    () =>
+      Object.fromEntries(
+        Object.entries(statsSpec).map(([pageName, pageStatSpec]) => [
+          pageName,
+          Object.fromEntries(
+            Object.entries(pageStatSpec).map(
+              ([categoryName, { label, categories }]) => [
+                categoryName,
+                {
+                  label,
+                  items: (
+                    categories as (
+                      backendStatResult:
+                        | BackendStatsResult[typeof categoryName]
+                        | undefined
+                    ) => StatCategoryReturn
+                  )(backEndResult?.[categoryName]),
+                },
+              ]
+            )
+          ),
+        ])
+      ),
+    [backEndResult]
+  );
+}
+
+function useDefaultStatsToAdd(
+  layout: {
+    readonly label: string;
+    readonly categories: RA<{
+      readonly label: string;
+      readonly items: RA<CustomStat | DefaultStat>;
+    }>;
+  },
+  defaultLayout: StatLayout
+): StatLayout {
+  return React.useMemo((): StatLayout => {
+    const listToUse = layout.categories.flatMap(({ items }) =>
+      items.filter((item): item is DefaultStat => item.type === 'DefaultStat')
+    );
+    return defaultLayout.map((defaultLayoutPage) => ({
+      label: defaultLayoutPage.label,
+      categories: defaultLayoutPage.categories
+        .map(({ label, items }) => ({
+          label,
+          items: items.filter(
+            (defaultItem) =>
+              defaultItem.type === 'DefaultStat' &&
+              !listToUse.some(
+                ({ pageName, categoryName, itemName }) =>
+                  pageName === defaultItem.pageName &&
+                  categoryName === defaultItem.categoryName &&
+                  itemName === defaultItem.itemName
+              )
+          ),
+        }))
+        .filter(({ items }) => items.length > 0),
+    }));
+  }, [layout, defaultLayout]);
+}
+
+export function StatsPage(): JSX.Element {
+  const [customLayout, setLayout] = usePref(
+    'statistics',
+    'appearance',
+    'layout'
+  );
+  const statsSpec = useStatsSpec();
+
+  const defaultLayout = useDefaultLayout(statsSpec);
+  const layout = customLayout ?? defaultLayout;
+  const [state, setState] = React.useState<
+    | State<
+        'EditingState',
+        {
+          readonly addingItem:
+            | {
+                readonly pageIndex: number;
+                readonly categoryIndex: number;
+              }
+            | undefined;
+        }
+      >
+    | State<'DefaultState'>
+  >({ type: 'DefaultState' });
+  const isEditing = state.type === 'EditingState';
+  const isAddingItem = isEditing && state.addingItem !== undefined;
+  const [activePageIndex, setActivePageIndex] = React.useState<number>(0);
+  const defaultStatsToAdd = useDefaultStatsToAdd(
+    layout[activePageIndex],
+    defaultLayout
+  );
+  console.log(defaultStatsToAdd);
```

Comment:
Remove console.log

Reply:
FYI: you can use logpoints instead of console.log - https://www.wisdomgeek.com/development/web-development/javascript/chrome-devtools-using-logpoints-for-logging-messages-directly/

Benefits:
- You can add/remove them at any point without reloading the page
- They are no part of the code, so you won't accidentally commit them
- At any point, you can convert them into a breakpoint or a conditional breakpoint